api/modules/projects/controller.js

```diff
@@ -154,6 +155,45 @@ class ProjectsController extends BaseController {
 
     return reply(result);
   }
+
+  renameFolder(request, reply) {
+    const projectId = request.pre.records.models[0].get(`id`);
+    const renamedPaths = request.payload;
+
+    function renameOperation(transaction) {
+      return Promise.map(Object.keys(renamedPaths), function(oldPath) {
+        return FilesModel.query({
+          where: {
+            path: oldPath,
+            project_id: projectId
+          }
+        })
+        .fetch({ transacting: transaction })
+        .then(function(file) {
+          if(!file) {
+            // We ignore file paths that were not found
+            // so that we do not fail the entire operation
+            // because of one bad value
+            return Promise.resolve();
+          }
+
+          file.set({ path: renamedPaths[oldPath] });
+
+          return file.save(file.changed, {
+            patch: true,
+            method: `update`,
+            transacting: transaction
+          });
+        });
+      }, { concurrency: 3 });
```

Comment:
coz reasons!!! I honestly chose a random number, feel free to suggest a way to select a number